client/server/pages/index.js

```diff
@@ -655,6 +658,13 @@ module.exports = function() {
 		JETPACK_CLOUD_SECTION_DEFINITION.paths.forEach( sectionPath =>
 			handleSectionPath( JETPACK_CLOUD_SECTION_DEFINITION, sectionPath, 'entry-jetpack-cloud' )
 		);
+
+		// catchall to render 404 for all routes not whitelisted in client/sections
+		app.use( render404( 'entry-jetpack-cloud' ) );
```

Comment:
Seems like the `render404` and `renderServerError` middlewares go together both here and when hooking to the main Calypso app. Should we abstract that in a helper for that to reduce repetition?

Reply:
Agreed, let me see if I can resolve that.

Reply:
Upon review, I actually think I'll defer. I think that passing in the `app` is messy, but I can't see a good way without combining the functions somehow.

I'll merge as-is, if that's okay :)